qiskit_nature/operators/second_quantization/star_algebra.py

```diff
@@ -43,9 +44,11 @@ def mul(self, other: complex):
         return NotImplementedError
 
     def __mul__(self, other: complex):
+        _warn()
```

Comment:
Shouldn't it be enough to do this once in the `__init__()` method?

Reply:
`StarAlgebra` does not have `__init__()`. First, I consider to use `__init_subclass__`, but this is dirty hack, so I decided to make as this.

Reply:
Well you could add an `__init__()` which throws the warning though, right?

Reply:
No, it doesn't make sense because the subclass doesn't always call it.